### DISCUSSION TITLE
Handling encrypted keys for ios.compare_config()

### DISCUSSION THREAD

**Discussion by cslayton65** [napalm-automation/napalm#2025] [Q&A] [answered]
I am using type 7 encryption for some keys and the keys will vary from device to device so that the ios.compare_config() does not match.  My workaround is to use the unencrypted key in the .cfg file which always gets committed using ios.commit_config().  Is there a better approach?

**Comment by ktbyers** [comment, marked as answer]
I would probably pick a given `type 7` key and just consistently use that (assume the actual key is the same for the devices).

Please test/verify this is right and make sure you have an alternate way to login to the device. I am pretty confident that will work properly, however.
